indico/MaKaC/common/search.py

```diff
@@ -30,21 +34,38 @@ def searchUsers(surName="", name="", organisation="", email="", conferenceId=Non
         # search users
         people = AvatarHolder().match(criteria, exact=exactMatch, searchInAuthenticators=searchExt)
 
-        # search authors
-        if conferenceId is not None:
-            conference = ConferenceHolder().getById(conferenceId)
-            authorIndex = conference.getAuthorIndex()
-            authors = authorIndex.match(criteria, exact=exactMatch)
-            # merge with users
-            users = people
-            people = []
-            emails = []
-            for user in users:
-                people.append(user)
-                emails.extend(user.getEmails())
-            for author in authors:
-                if author.getEmail() not in emails:
-                    people.append(author)
         return people
     else:
         return []
+
+
+def getAuthorsFromAuthorIndex(event, limit):
+    author_index = event.getAuthorIndex()
+    fav_users_emails = {email for emails in (f.all_emails for f in session.user.favorite_users) for email in emails}
+    sorted_authors = sorted(((len(values), values[0]) for author, values in author_index.iteritems()
+                            if values[0].getEmail() not in fav_users_emails), reverse=True)
+    authors = []
+    # Check if there is a corresponding Avatar for the specific user and use that.
+    for author in [author for contributions, author in sorted_authors[:limit]]:
+        avatar = AvatarHolder().match({"email": author.getEmail()}, searchInAuthenticators=False, exact=True)
```

Comment:
Since we are at it, why not use the [new `search_users(...)` function](https://github.com/indico/indico/blob/master/indico/modules/users/util.py#L100)?